app/controllers/foreman_ansible/api/v2/hostgroups_controller_extensions.rb

```diff
@@ -45,10 +47,63 @@ def multiple_play_roles
 
             render_message @result
           end
+
+          api :GET, '/hostgroups/:id/list_ansible_roles',
+              N_('Lists assigned Ansible roles')
+          param :id, :identifier, :required => true
+
+          def list_ansible_roles
+            find_resource
```

Comment:
Best to summarize this at the beginning of the file as `before_action :find_resource, :only => [:play_roles, :list_ansible_roles, :ansible_roles]` and remove all of the `find_resource` in the methods